### PR TITLE
feat: Add healing trace

### PR DESCRIPTION
**Pull request by klauspost**
## Description

Adds version healing traces.

Requires https://github.com/minio/madmin-go/pull/122

## How to test this PR?

Use mc PR: https://github.com/minio/mc/pull/4229

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: cmd/erasure-healing.go [hunk 1]
```diff
@@ -285,6 +285,29 @@ func (er erasureObjects) healObject(ctx context.Context, bucket string, object s
...
+			}
+			if versionID != "" && versionID != "null" {
+				tr.Path += " v-" + versionID
+			}
+			if err != nil {
+				tr.Error = err.Error()
+			} else {
+				tr.HealResult = &result
+			}
+			globalTrace.Publish(tr)
```
Comment (vadmeste) [inline cmd/erasure-healing.go]: Maybe let's move this to another function

**Review by vadmeste — COMMENTED** [review]
One comment

File: cmd/erasure-healing.go (hunk 1 above)
Comment (klauspost) [reply cmd/erasure-healing.go]: Let me give it a try. Not sure there is much to reuse.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#02231968-a223-4fe0-9ad7-34e858f243af) |
| mint-fs.sh | :x: [more...](#1504f358-5ec8-4bc7-a2f3-5a4e6e5d52bc) |
| mint-erasure.sh | :x: [more...](#8bc62e8c-9474-40b6-bdeb-4df8fdc7f5ef) |
| mint-dist-erasure.sh | :x: [more...](#199fed5f-2996-464f-a836-d8aec12966c4) |
| mint-compress-encrypt-dist-erasure.sh | :x: [more...](#99a51ce3-7590-47fe-a961-25fbfbd1e417) |
| mint-pools.sh | :x: [more...](#6b4eb0ed-6a34-4ae7-b90c-e8344f1aa016) |
<a name="6b4eb0ed-6a34-4ae7-b90c-e8344f1aa016"></a>
### 15616-658cf3c/mint-pools.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s3:30522
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 6 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 45 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 57 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 34 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 18 seconds
(9/14) Running minio-java tests ... FAILED in 5 minutes and 30 seconds
Aug 30, 2022 10:37:20 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-2pb17bc</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-2pb17bc</Key><ETag>&#34;1a4e41d84fde7264e9ef1a09c409dd0c-3&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:21 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-6entth</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-6entth</Key><ETag>&#34;1a4e41d84fde7264e9ef1a09c409dd0c-3&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-f0r09b</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-f0r09b</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-2cl7a8l</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-2cl7a8l</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-1j08rag</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-1j08rag</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-2s0d2ui</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-2s0d2ui</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-ega0k5</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-ega0k5</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-1epk4it</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-1epk4it</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:22 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-50tsbn</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-50tsbn</Key><ETag>&#34;901a15da453093fec88d9f69bbe6cd8d-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:23 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-vfgm8u</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-vfgm8u</Key><ETag>&#34;b89bd676225da2a904a2fc51cc75a0fe-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
Aug 30, 2022 10:37:24 AM io.minio.S3Base lambda$completeMultipartUploadAsync$12
WARNING: S3 service returned unknown XML for CompleteMultipartUpload REST API. <?xml version="1.0" encoding="UTF-8"?>
<CompleteMultipartUploadResult xmlns="http://s3.amazonaws.com/doc/2006-03-01/"><Location>http://minio-k8s3:30522/minio-java-test-3tibsqp/minio-java-test-3lvuphh</Location><Bucket>minio-java-test-3tibsqp</Bucket><Key>minio-java-test-3lvuphh</Key><ETag>&#34;b89bd676225da2a904a2fc51cc75a0fe-2&#34;</ETag><ChecksumCRC32></ChecksumCRC32><ChecksumCRC32C></ChecksumCRC32C><ChecksumSHA1></ChecksumSHA1><ChecksumSHA256></ChecksumSHA256></CompleteMultipartUploadResult>
(9/14) Running minio-js tests ... done in 1 minutes and 36 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 30 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 5 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
<a name="8bc62e8c-9474-40b6-bdeb-4df8fdc7f5ef"></a>
### 15616-658cf3c/mint-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30963
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 21 seconds
(8/14) Running minio-go tests ... done in 55 seconds
(9/14) Running minio-java tests ... done in 22 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 37 seconds
{
  "name": "minio-js",
  "function": "listenBucketNotification(bucketName, prefix, suffix, events)",
  "args": "bucketName:minio-js-test-4c64ba58-f0c5-4eb1-a1a9-04f737ac9066, events:s3:ObjectRemoved:*",
  "duration": 118368,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 50 seconds
(11/14) Running s3cmd tests ... done in 24 seconds
(12/14) Running s3select tests ... done in 2 seconds
(13/14) Running versioning tests ... done in 3 minutes and 2 seconds

Executed 13 out of 14 tests successfully.
```
<a name="199fed5f-2996-464f-a836-d8aec12966c4"></a>
### 15616-658cf3c/mint-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s6:30621
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 1 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 56 seconds
(6/14) Running healthcheck tests ... done in 1 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 37 seconds
(9/14) Running minio-java tests ... FAILED in 2 minutes and 7 seconds
{
  "name": "minio-java",
  "function": "listObjects()",
  "args": "[bucket, prefix, recursive, 1050 objects]",
  "duration": 120425,
  "status": "FAIL",
  "error": "error occurred\nErrorResponse(code = BucketNotEmpty, message = The bucket you tried to delete is not empty, bucketName = minio-java-test-3nnrcaq, objectName = null, resource = /minio-java-test-3nnrcaq, requestId = 171019303C4963D6, hostId = 34148ab5-864c-4cee-8566-4f5d6e15d1b4)\nrequest={method=DELETE, url=http://minio-k8s6:30621/minio-java-test-3nnrcaq, headers=Host: minio-k8s6:30621\nAccept-Encoding: identity\nUser-Agent: MinIO (Linux; amd64) minio-java/8.4.3\nContent-MD5: 1B2M2Y8AsgTpgAmY7PhCfg==\nx-amz-content-sha256: e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855\nx-amz-date: 20220830T103917Z\nAuthorization: AWS4-HMAC-SHA256 Credential=*REDACTED*/20220830/us-east-1/s3/aws4_request, SignedHeaders=content-md5;host;x-amz-content-sha256;x-amz-date, Signature=*REDACTED*\n}\nresponse={code=409, headers=Accept-Ranges: bytes\nContent-Length: 328\nContent-Security-Policy: block-all-mixed-content\nContent-Type: application/xml\nServer: MinIO\nStrict-Transport-Security: max-age=31536000; includeSubDomains\nVary: Origin\nVary: Accept-Encoding\nX-Amz-Request-Id: 171019303C4963D6\nX-Content-Type-Options: nosniff\nX-Xss-Protection: 1; mode=block\nDate: Tue, 30 Aug 2022 10:39:17 GMT\n}\n >>> [io.minio.S3Base$1.onResponse(S3Base.java:690), okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519), java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128), java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628), java.base/java.lang.Thread.run(Thread.java:829)]"
}
(9/14) Running minio-js tests ... done in 1 minutes and 4 seconds
(10/14) Running minio-py tests ... done in 1 minutes and 16 seconds
(11/14) Running s3cmd tests ... done in 26 seconds
(12/14) Running s3select tests ... done in 4 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
<a name="99a51ce3-7590-47fe-a961-25fbfbd1e417"></a>
### 15616-658cf3c/mint-compress-encrypt-dist-erasure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s1:30471
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 6 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 44 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 51 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 18 seconds
(9/14) Running minio-java tests ... done in 31 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 37 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 3,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 20 seconds
(11/14) Running s3cmd tests ... done in 25 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 4 seconds

Executed 13 out of 14 tests successfully.
```
<a name="1504f358-5ec8-4bc7-a2f3-5a4e6e5d52bc"></a>
### 15616-658cf3c/mint-fs.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s5:30933
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 1 seconds
(3/14) Running aws-sdk-php tests ... done in 42 seconds
(4/14) Running aws-sdk-ruby tests ... done in 4 seconds
(5/14) Running awscli tests ... done in 51 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 27 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 27 seconds
(9/14) Running minio-java tests ... done in 19 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 37 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 3,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 52 seconds
(11/14) Running s3cmd tests ... done in 23 seconds
(12/14) Running s3select tests ... done in 1 seconds
(13/14) Running versioning tests ... done in 1 minutes and 1 seconds

Executed 13 out of 14 tests successfully.
```
<a name="02231968-a223-4fe0-9ad7-34e858f243af"></a>
### 15616-658cf3c/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-k8s8:31407
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0
RUN_ON_FAIL:          0

To get logs, run 'docker cp :/mint/log /tmp/mint-logs'

(1/14) Running aws-sdk-go tests ... done in 8 seconds
(2/14) Running aws-sdk-java tests ... done in 2 seconds
(3/14) Running aws-sdk-php tests ... done in 43 seconds
(4/14) Running aws-sdk-ruby tests ... done in 5 seconds
(5/14) Running awscli tests ... done in 53 seconds
(6/14) Running healthcheck tests ... done in 0 seconds
(7/14) Running mc tests ... done in 26 seconds
(8/14) Running minio-go tests ... done in 1 minutes and 12 seconds
(9/14) Running minio-java tests ... done in 33 seconds
(10/14) Running minio-js tests ... FAILED in 2 minutes and 38 seconds
{
  "name": "minio-js",
  "function": "\"after all\" hook in \"functional tests\"",
  "duration": 2,
  "status": "FAIL",
  "error": "Error: read ECONNRESET at TCP.onStreamRead (internal/stream_base_commons.js:209:20)"
}
(10/14) Running minio-py tests ... done in 1 minutes and 24 seconds
(11/14) Running s3cmd tests ... done in 35 seconds
(12/14) Running s3select tests ... done in 3 seconds
(13/14) Running versioning tests ... done in 3 minutes and 5 seconds

Executed 13 out of 14 tests successfully.
```
Deleting image on docker hub
Deleting image locally

File: cmd/erasure-healing.go (hunk 1 above)
Comment (klauspost) [reply cmd/erasure-healing.go]: Works fine. Adding to `healBucket` as well.

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

File: go.mod [hunk 2]
```diff
@@ -222,3 +222,5 @@ require (
 	gopkg.in/ini.v1 v1.66.6 // indirect
 	gopkg.in/square/go-jose.v2 v2.5.1 // indirect
 )
+
+replace github.com/minio/madmin-go => github.com/klauspost/madmin-go v1.0.15-0.20220830094240-17a19a73d7b8
```
Comment (harshavardhana) [inline go.mod]: madmin-go released may need to upgrade to that version.

**Review by harshavardhana — CHANGES_REQUESTED** [review]

File: go.mod (hunk 2 above)
Comment (klauspost) [reply go.mod]: Done

**Review by harshavardhana — APPROVED** [review]

**Review by vadmeste — APPROVED** [review]
LGTM

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Pull request merged**